subsys/pm/policy.c

```diff
@@ -66,26 +66,26 @@ static int64_t next_event_cyc = -1;
 /** @brief Update maximum allowed latency. */
 static void update_max_latency(void)
 {
-	int32_t new_max_latency_us = SYS_FOREVER_US;
+	uint32_t new_max_latency_us = SYS_FOREVER_US;
 	struct pm_policy_latency_request *req;
 
 	SYS_SLIST_FOR_EACH_CONTAINER(&latency_reqs, req, node) {
 		if ((new_max_latency_us == SYS_FOREVER_US) ||
-		    ((int32_t)req->value_us < new_max_latency_us)) {
-			new_max_latency_us = (int32_t)req->value_us;
+		    (req->value_us < new_max_latency_us)) {
```

Comment:
`(new_max_latency_us == SYS_FOREVER_US)` will be true the first time of this loop and consequently `new_max_latency_us` will be updated.

Reply:
I think I need to update the information on the pull request, because it doesn't accurately describe the issue. To elaborate:
- Thread "A" creates a request with latency in the range [0, `INT32_MAX`] for the time that it is running some operations, and `SYS_FOREVER_US` otherwise
- Thread "B" creates a request with latency in the range [0, `INT32_MAX`] for the time that it is running some operations, and `SYS_FOREVER_US` otherwise

The problem happens in the following situation:
- "A" is running operations and "B" is not, so latency request for "A" is suppose 0 and latency request for "B" is SYS_FOREVER_US (i.e., -1)
- When `update_max_latency()` runs:
    - latency request for "A" is 0, so `new_max_latency_us` will be updated to 0 since new_max_latency_us is initialized to `SYS_FOREVER_US`
    - latency request for "B" is -1, so `new_max_latency_us` will be updated to -1 since -1 is less that 0

This results in the latency request for "A" not being honoured.